.github/workflows/notebook-test.yml

```diff
@@ -0,0 +1,67 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+name: Test notebooks
+on:
+  pull_request:
+    paths:
+      - "docs/**/*.ipynb"
+      - "!docs/api/**/*"
+  workflow_dispatch:
+jobs:
+  execute:
+    name: Execute notebooks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: "pip"
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@af2816c65436325c50621100d67f6e853cd1b0f1
+
+      - name: Install LaTeX dependencies
+        run: |
+          # Add your notebook to this list if it needs latex to run
+          NEEDS_LATEX="docs/build/circuit-visualization.ipynb"
+          for FILE in $NEEDS_LATEX; do
+            if [[ "${{ steps.changed-files.outputs.all_changed_files }}" = *"$FILE"* ]]; then
+              sudo apt-get install texlive-pictures texlive-latex-extra poppler-utils
+              break
+            fi
+          done
```

Comment:
The latex packages are pretty large and only used by one cell. Check out these commits' runs to see this step working:
- Skip installation https://github.com/Qiskit/documentation/pull/483/commits/1d6872fc77eb27a36c0c0e320fc5066c9356989d
- Install texlive https://github.com/Qiskit/documentation/pull/483/commits/0f386392e62cd21c7a42519efb8e690f9cbed867